### PR TITLE
issue-892 Find closest number in array

### PR DESCRIPTION
**Pull request by rray14**
Issue #892 
Finding the closest number in an array

### CODE REVIEW

File: Code/C++/closest_number_in_array.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    if(target<=arr[0]){
+        return arr[0];
+    }
+    else if(target>=arr[n-1]){
+        return arr[n-1];
+    }
+    
+    //Iterative binary search
+    while(low<high){
+        mid=(low+high)/2;
```
Comment (tarun26091999) [inline Code/C++/closest_number_in_array.cpp:42]: this is wrong

File: Code/C++/closest_number_in_array.cpp [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+            return arr[mid];
+
+        //If target is less than middle element, then search left portion
+        if(arr[mid]>target){
+
+            //If target lies between arr[mid] and arrr[mid-1], take min of the two
+            if(mid>0 && arr[mid-1]<target)
+                return min(arr[mid-1],arr[mid],target);
+            //else
+            high=mid;
```
Comment (tarun26091999) [inline Code/C++/closest_number_in_array.cpp:53]: check this also

**Review by tarun26091999 — CHANGES_REQUESTED** [review]
also link to appropriate folder.

File: Code/C++/closest_number_in_array.cpp (hunk 2 above)
Comment (rray14) [reply Code/C++/closest_number_in_array.cpp:53]: sir @tarun26091999 but the code is working and giving the result though, what exactly is wrong here, sorry am I missing something.

**Pull request closed without merging**